core/src/test/scala/org/apache/spark/scheduler/HealthTrackerSuite.scala

```diff
@@ -554,6 +554,51 @@ class HealthTrackerSuite extends SparkFunSuite with BeforeAndAfterEach with Mock
     verify(allocationClientMock).killExecutorsOnHost("hostA")
   }
 
+  test("excluding decommission and kills executors when enabled") {
+    val allocationClientMock = mock[ExecutorAllocationClient]
+    healthTracker = new HealthTracker(listenerBusMock, conf, Some(allocationClientMock), clock)
+
+    // verify we decommission when configured
+    conf.set(config.EXCLUDE_ON_FAILURE_KILL_ENABLED, true)
+    conf.set(config.DECOMMISSION_ENABLED.key, "true")
```

Comment:
this conf is not required?

Reply:
Without decommissioning enabled the executor _could_ do something different (although right now your right this is not needed). It just seemed like a better future proof configuration to have it on.